lib/dynflow/dispatcher/client_dispatcher.rb

```diff
@@ -26,16 +26,84 @@ def success!(resolve_to)
         end
       end
 
+      # Class used for reducing the number of sent Pings among worlds.
+      # World's coordinator record include the time when was the world
+      # seen for the last time. This class can be used to query this
+      # information and determine whether the record is "fresh enough"
+      # or whether the Ping really needs to be sent.
+      class PingCache
+        # Format string used for formating and parsing times
+        TIME_FORMAT = '%Y-%m-%d %H:%M:%S.%L'.freeze
+
+        # Maximum age of a record in seconds, older records are not considered fresh
+        PING_CACHE_AGE = 10
```

Comment:
With the light of https://bugzilla.redhat.com/show_bug.cgi?id=1538688, I wonder if we could increase this to something like 70, and do some heartbeat internally to update the ping cache when the executor is live, so that we would avoid the polling completely, which might be an issue when the executor is too busy.